### DISCUSSION TITLE
Welcome to pyModeS Discussions!

### DISCUSSION THREAD

**Discussion by junzis** [junzis/pyModeS#88] [General]
## 👋 Welcome to pyModeS discussion board

We’re using Discussions as a place to connect with other members of our community.  

(For bug report, please keep using *issues* board)

We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others, and be open-minded. 

To get started, feel free to comment below with an introduction of yourself, or tell us your story of using pyModeS.

**Comment by fra-fdg** [comment]
Hello Junzis,

First many thanks for all of your work.

None or my recieved DF4 / DF5 messages return 0 the crc().
Additionnaly these messages are frequently inconsistant.

Does the crc() function suitable for short message ?
If not is there an other way to check validity if short messages ?

Thanks in advance.

François

**Reply by junzis** [reply]
CRC will only yeilds 0 for correct DF 11, 17 and 18  messages. 

For other types of Mode S messages, it is a bit more complicated. Have a look at (https://mode-s.org/decode/content/mode-s/1-basics.html)

**Comment by fraciocca** [comment]
Hi Junzis,

First of all really thanks for your work.
I was trying to use your code as I wanted to decode ADS-B messages directly received by a Transponder.
Firstly I want to ask you if doing such a thing is possible with your code.
Moreover, I can't install your library on Python 3.7 on my Mac, in fact when I write the following instruction on Python terminal (just installed) pip install pyModeS, it says me invalid syntax.
Would you kindly tell me if there is anything wrong that I'm missing?
Thank you for any answer
Best
Francesco

**Reply by xoolive** [reply]
This error sounds like your pip points to Python 2. Try `python3 -m pip install etc.`

**Reply by fraciocca** [reply]
Hi Xavier thank you for your answer.
However it still gives me an invalid Syntax error on pip...

**Reply by xoolive** [reply]
Well, I am on a Mac too and I can't reproduce... Sorry then

**Comment by dsmith636** [comment]
I tried to install pip pyModeS and my ubuntu machine didn"t like it, it only installed in pipx.  I wanted to call it in a python script with an import command but never had success.    It installs in pipx as modeslive and I hunted down the install folder and made it run on command from the bin folder it installed in.  Just could not call it from a script
